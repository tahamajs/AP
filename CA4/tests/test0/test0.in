report_salaries
report_employee_salary 2
report_team_salary 1
update_team_bonus 2 10
add_working_hours 6 1 17 19
show_salary_config junior
update_team_bonus 1 15
report_team_salary 1
report_employee_salary 5
report_team_salary 2
show_salary_config manager
update_salary_config team_lead - - 140 225 -
report_total_hours_per_day 8 13
add_working_hours 1 3 7 16
delete_working_hours 2 0
report_employee_salary 1
add_working_hours 1 12 12 23
report_employee_salary 3
report_employee_per_hour 4 25
report_employee_salary 1
add_working_hours 1 27 8 18
report_team_salary 0
report_team_salary 2
delete_working_hours 1 8
report_salaries
report_total_hours_per_day 19 5
add_working_hours 4 12 4 6
report_employee_salary 4
report_employee_salary 1
report_team_salary 1
delete_working_hours 3 8
report_employee_salary 3
add_working_hours 5 1 18 22
report_employee_salary 5
report_employee_salary 8
