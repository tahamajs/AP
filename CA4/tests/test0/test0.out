ID: 1
Name: AzadehZahedi
Total Working Hours: 200
Total Earning: 21250
---
ID: 2
Name: GolnazEmadi
Total Working Hours: 217
Total Earning: 53761
---
ID: 3
Name: ParvinKarami
Total Working Hours: 228
Total Earning: 54857
---
ID: 4
Name: MehranAmiri
Total Working Hours: 226
Total Earning: 46749
---
ID: 5
Name: BehzadAbbasi
Total Working Hours: 175
Total Earning: 28080
---
ID: 2
Name: GolnazEmadi
Age: 39
Level: team_lead
Team ID: 1
Total Working Hours: 217
Absent Days: 2
Salary: 79060
Bonus: 0
Tax: 25299
Total Earning: 53761
ID: 1
Head ID: 2
Head Name: GolnazEmadi
Team Total Working Hours: 443
Average Member Working Hours: 221.5
Bonus: 0
---
Member ID: 2
Total Earning: 53761
---
Member ID: 4
Total Earning: 46749
---
OK
EMPLOYEE_NOT_FOUND
Base Salary: 13000
Salary Per Hour: 60
Salary Per Extra Hour: 48
Official Working Hours: 210
Tax: 15%
OK
ID: 1
Head ID: 2
Head Name: GolnazEmadi
Team Total Working Hours: 443
Average Member Working Hours: 221.5
Bonus: 15
---
Member ID: 2
Total Earning: 61825
---
Member ID: 4
Total Earning: 53761
---
ID: 5
Name: BehzadAbbasi
Age: 26
Level: senior
Team ID: N/A
Total Working Hours: 175
Absent Days: 4
Salary: 39000
Bonus: 0
Tax: 10920
Total Earning: 28080
ID: 2
Head ID: 3
Head Name: ParvinKarami
Team Total Working Hours: 428
Average Member Working Hours: 214.0
Bonus: 10
---
Member ID: 1
Total Earning: 23375
---
Member ID: 3
Total Earning: 60343
---
INVALID_LEVEL
OK
Day #8: 34
Day #9: 31
Day #10: 37
Day #11: 43
Day #12: 21
Day #13: 40
---
Day(s) with Max Working Hours: 11
Day(s) with Min Working Hours: 12
OK
INVALID_ARGUMENTS
ID: 1
Name: AzadehZahedi
Age: 28
Level: junior
Team ID: 2
Total Working Hours: 209
Absent Days: 6
Salary: 25540
Bonus: 2554
Tax: 4214
Total Earning: 23880
OK
ID: 3
Name: ParvinKarami
Age: 45
Level: team_lead
Team ID: 2
Total Working Hours: 228
Absent Days: 2
Salary: 80920
Bonus: 8092
Tax: 28484
Total Earning: 60528
INVALID_ARGUMENTS
ID: 1
Name: AzadehZahedi
Age: 28
Level: junior
Team ID: 2
Total Working Hours: 220
Absent Days: 5
Salary: 26080
Bonus: 2608
Tax: 4303
Total Earning: 24385
OK
TEAM_NOT_FOUND
ID: 2
Head ID: 3
Head Name: ParvinKarami
Team Total Working Hours: 458
Average Member Working Hours: 229.0
Bonus: 10
---
Member ID: 1
Total Earning: 24834
---
Member ID: 3
Total Earning: 60528
---
OK
ID: 1
Name: AzadehZahedi
Total Working Hours: 226
Total Earning: 24654
---
ID: 2
Name: GolnazEmadi
Total Working Hours: 217
Total Earning: 61825
---
ID: 3
Name: ParvinKarami
Total Working Hours: 228
Total Earning: 60528
---
ID: 4
Name: MehranAmiri
Total Working Hours: 226
Total Earning: 53761
---
ID: 5
Name: BehzadAbbasi
Total Working Hours: 175
Total Earning: 28080
---
INVALID_ARGUMENTS
INVALID_INTERVAL
ID: 4
Name: MehranAmiri
Age: 32
Level: expert
Team ID: 1
Total Working Hours: 226
Absent Days: 0
Salary: 62332
Bonus: 9350
Tax: 17921
Total Earning: 53761
ID: 1
Name: AzadehZahedi
Age: 28
Level: junior
Team ID: 2
Total Working Hours: 226
Absent Days: 5
Salary: 26368
Bonus: 2637
Tax: 4351
Total Earning: 24654
ID: 1
Head ID: 2
Head Name: GolnazEmadi
Team Total Working Hours: 443
Average Member Working Hours: 221.5
Bonus: 15
---
Member ID: 2
Total Earning: 61825
---
Member ID: 4
Total Earning: 53761
---
OK
ID: 3
Name: ParvinKarami
Age: 45
Level: team_lead
Team ID: 2
Total Working Hours: 221
Absent Days: 3
Salary: 79780
Bonus: 7978
Tax: 28083
Total Earning: 59675
OK
ID: 5
Name: BehzadAbbasi
Age: 26
Level: senior
Team ID: N/A
Total Working Hours: 179
Absent Days: 4
Salary: 39400
Bonus: 0
Tax: 11032
Total Earning: 28368
EMPLOYEE_NOT_FOUND
